### PR TITLE
feat(test): add nvim-coverage

### PR DESCRIPTION
**Pull request by 0xricksanchez**

### CODE REVIEW

File: lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua [hunk 1]
```diff
@@ -0,0 +1,8 @@
+return {
+  "andythigpen/nvim-coverage",
+  lazy = false,
```
Comment (owittek) [inline lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua:3]:
```suggestion
  event = "User AstroFile",
```

or check for a filetypes with `ft = { "..." }` or load the plugin when [running a command](https://github.com/andythigpen/nvim-coverage/blob/e542b6afbc7d92acd6502e15124ca2f376964b76/lua/coverage/init.lua#L18). Just disabling lazy-loading has performance implications for startup when the plugin is not actually needed.

File: lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  requires = { "nvim-lua/plenary.nvim" },
+  -- Optional: needed for PHP when using the cobertura parser
+  rocks = { "lua-xmlreader" },
+  config = function() require("coverage").setup() end,
```
Comment (owittek) [inline lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua:7]:
```suggestion
```

Lazy automaticially calls the setup function even if `config` is not defined. [See here](https://github.com/folke/lazy.nvim#-plugin-spec)

**Review by owittek — CHANGES_REQUESTED** [review]

File: lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua [hunk 3]
```diff
@@ -0,0 +1,8 @@
+return {
+  "andythigpen/nvim-coverage",
+  lazy = false,
+  requires = { "nvim-lua/plenary.nvim" },
+  -- Optional: needed for PHP when using the cobertura parser
+  rocks = { "lua-xmlreader" },
```
[the remark this answers is not in the document]
Comment (0xricksanchez) [reply lua/astrocommunity/test/nvim-coverage/nvim-coverage.lua:6]: Oh right, that's a packer thing.. I guess leaving php out is fine for now anyway.

**Review by owittek — APPROVED** [review]
LGTM

**Comment by owittek** [comment]
@AstroNvim/astrocommunity-maintainers need an approval to merge since I updated the branch.

**Review by mehalter — APPROVED** [review]
LGTM

**Pull request merged**